src/components/FolderCreationModal.jsx

```diff
@@ -54,17 +56,29 @@ const FolderCreationModal = ({
     }
   ]
 
-  const baseApiUrl = `${process.env.REACT_APP_BACKEND_URL}/sites/${siteName}${parentFolder ? `/collections/${parentFolder}` : ''}`
 
-  const saveHandler = async () => {
-    // TODO
-    console.log('saving')
-  }
+  const { mutate: saveHandler, isLoading, error: mutateError } = useMutation(
```

Comment:
As opposed to using `isLoading`, try using `mutateAsync` to replace `mutate` - that should return a promise which can be awaited, which is probably compatible with our `await callback()` in `LoadingButton`

Reply:
Previously, the `LoadingButton` stored its own loading state, which is why we needed to `await` the callback. We found this unnecessarily complicated and removed the state from the loading button by creating a separate `LoadingButtonReactQuery` so that the button component itself doesn't store any state but rather accepts an `isLoading` prop which determines the loading state of the button

Reply:
As discussed offline, moving forward we will be using `LoadingButton` instead of `LoadingButtonReactQuery`. Fixed in ce4a432